ray-on-gke/tpu/kuberay-tpu-webhook/webhook_main_test.go

```diff
@@ -4,34 +4,39 @@ import (
 	"encoding/json"
 	"errors"
 	"fmt"
+	"sort"
 	"strings"
 	"testing"
 
 	rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
 	"github.com/ray-project/kuberay/ray-operator/controllers/ray/utils"
+	"github.com/stretchr/testify/assert"
 	admissionv1 "k8s.io/api/admission/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
 	"k8s.io/utils/pointer"
-
-	"github.com/stretchr/testify/assert"
 )
 
 var (
-	namespaceStr        string
-	instanceName        string
-	groupNameStr        string
-	headGroupNameStr    string
-	testCPUWorker       *corev1.Pod
-	testTPUWorker       *corev1.Pod
-	testCPUPods         []*corev1.Pod
-	testTPUPods         []*corev1.Pod
-	testAdmissionReview *admissionv1.AdmissionReview
-	testRayCluster      *rayv1.RayCluster
-	headNodeIP          string
-	testWorkerGroupSpec *rayv1.WorkerGroupSpec
+	namespaceStr           string
+	instanceName           string
+	groupNameStr           string
+	headGroupNameStr       string
+	testCPUWorker          *corev1.Pod
+	testTPUWorker          *corev1.Pod
+	testCPUPods            []*corev1.Pod
+	testTPUPods            []*corev1.Pod
+	testInterceptedTPUPods []*corev1.Pod
+	testAdmissionReview    *admissionv1.AdmissionReview
+	testRayCluster         *rayv1.RayCluster
+	headNodeIP             string
+	testWorkerGroupSpec    *rayv1.WorkerGroupSpec
 )
 
 func setupTest(t *testing.T) {
```

Comment:
This function is setting global variables but runs for every test, even when the test doesn't use all the global variables being set here. Consider defining local variables within each test that needs to reference fake objects only scoped to that test

Reply:
Done in [b70535d](https://github.com/GoogleCloudPlatform/ai-on-gke/pull/723/commits/b70535d96c495b0afafbfe07cd8bfa82a3d0d5b8), I refactored the tests to build these objects in their own functions, and then only created them in the tests they were actually used. There are no global vars in the unit tests anymore.